### PR TITLE
feat(multiple-source-provider): add a multiple source provider

### LINKED ISSUE #786: As a <User>, I want to <get overlays from N sources>, so that <I can analyze different overlays>
* [x] @idanbarak @carmelorr define the configuration limit's (And, Or, Time, Sensor Type)
* [x] @idanbarak @carmelorr define the provider behavior (one fails other success?, one timeout other success?, etc.)
* [x] Design a provider of providers (gets N providers via injection and handles them)
* [x] Implement a provider of providers (gets N "Idahoo" providers (try different tokens) via injection and handle them via configuration)

### PR DESCRIPTION
**Pull request by AmitMY**
resolve #786

### CODE REVIEW

File: src/app/app/app-providers/overlay-source-providers/multiple-source-provider.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+								const dateObj = {
+									start: date.start ? new Date(date.start) : null,
+									end: date.end ? new Date(date.end) : null,
+								};
+								blackFilters.push({ sensor, timeRange: dateObj, coverage: this.coverageToFeature(polygon) });
+							});
+						});
+					});
+				});
+
```
Comment (yuvalbl) [inline src/app/app/app-providers/overlay-source-providers/multiple-source-provider.ts]: add documentation

File: src/app/app/app-providers/overlay-source-providers/multiple-source-provider.ts [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+		const mergedSortedOverlays = Promise.all(this.sourceConfigs
+			.map(s => s.provider.fetchMultiple(fetchParams, s.filters).toPromise()))
+			.then(overlaysArr => overlaysArr.reduce((a, b) => a.concat(b), []))
+			.then(overlays => overlays.sort((o1, o2) => o1.date > o2.date ? -1 : 1));
+
+		return Observable.from(mergedSortedOverlays);
+	}
+
+	public getStartDateViaLimitFacets(params: { facets, limit, region }): Observable<Array<Overlay>> {
+		// TODO
```
Comment (yuvalbl) [inline src/app/app/app-providers/overlay-source-providers/multiple-source-provider.ts]: rm

File: src/app/packages/overlays/models/base-overlay-source-provider.model.ts [hunk 3]
```diff
@@ -1,20 +1,96 @@
...
+				}
+				return area(intersection) > 0;
+			})
+			.filter(f => Boolean(timeIntersection(fetchParamsTimeRange, f.timeRange)))
+			.map(f => {
+				const region = intersect(f.coverage, regionFeature).geometry;
+
+				let newFetchParams: IFetchParams = {
+					region,
+					timeRange: fetchParams.timeRange
```
Comment (yuvalbl) [inline src/app/packages/overlays/models/base-overlay-source-provider.model.ts]: intersection

**Comment by AmitMY** [comment]
@asafMasa This PR is done.
You are welcomed to go over it, but prepare for a real headache.

It does not support all of the functionalities of a source provider, like:
https://github.com/AnSyn/ansyn/pull/833/files#diff-3b304f8fd2065e7b7c880893caf4d4cbR117
and
https://github.com/AnSyn/ansyn/pull/833/files#diff-3b304f8fd2065e7b7c880893caf4d4cbR130

as those would need some other refactoring, and the current state is more than workable. 
I urge you to merge it, as it has all of the functionality one needs for developing for the client.

File: src/app/app/app-providers/overlay-source-providers/multiple-source-provider.ts [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+				this.sourceConfigs.push({
+					provider,
+					filters: whiteFilters
+				});
+			}
+		});
+	}
+
+	public getById(id: string): Observable<Overlay> {
+		throw new Error('need more info, id is insufficient');
```
Comment (asafMasa) [inline src/app/app/app-providers/overlay-source-providers/multiple-source-provider.ts]: Need to support Get By Id

File: src/app/app/app-providers/overlay-source-providers/idaho-source-provider2.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { Inject, Injectable } from '@angular/core';
```
Comment (asafMasa) [inline src/app/app/app-providers/overlay-source-providers/idaho-source-provider2.ts]: This file is good for **tests** not for production, please remove "IDAHOO 2" from code and facets after approving with designer

File: src/app/app/app-providers/overlay-source-providers/multiple-source-provider.ts (hunk 4 above)
Comment (AmitMY) [reply src/app/app/app-providers/overlay-source-providers/multiple-source-provider.ts]: With what logic?

**Pull request merged**